libp2p/stream_muxer/mplex/mplex.py

```diff
@@ -104,9 +109,13 @@ def _get_next_channel_id(self) -> int:
         return next_id
 
     async def _initialize_stream(self, stream_id: StreamID, name: str) -> MplexStream:
-        stream = MplexStream(name, stream_id, self)
+        # Use an unbounded buffer, to avoid `handle_incoming` being blocked when doing
+        # `send_channel.send`.
```

Comment:
I really don't think this is a good idea.  It will hide a real problem by allowing unbounded memory growth...

Reply:
Do you think it makes sense to use a bounded buffer for `streams_msg_channels`, i.e. `trio.open_memory_channel[bytes](BUFFER_SIZE)`? Or do you expect all the channels to be unbuffered, i.e. `trio.open_memory_channel[bytes](0)`?

Reply:
Bounded to some reasonable upper bound seems fine.  I don't have context to know what an appropriate value would be here but it's easy to change so no need to stress over it too much.

Reply:
I've changed the sizes of
- `Mplex.new_stream_send_channel` to `0`. Its consumer is `SwarmConn`, which should be able to consume the data pretty soon in another task.
https://github.com/libp2p/py-libp2p/blob/5b03a7ad9f801fe7fa26c70515dadbe2e9bf799e/libp2p/stream_muxer/mplex/mplex.py#L70-L71
- `Mplex.streams_msg_channels` to `MPLEX_MESSAGE_CHANNEL_SIZE = 8`, conforming to the go implementation. If the channel is full and new messages arrive, the stream is reset.
https://github.com/libp2p/py-libp2p/blob/5b03a7ad9f801fe7fa26c70515dadbe2e9bf799e/libp2p/stream_muxer/mplex/mplex.py#L113-L115
- `Pubsub.subscribed_topics_send[topic]` to `SUBSCRIPTION_CHANNEL_SIZE = 32`, conforming to the go implementation. If the channel is full and new messages arrive, the messages are just dropped.
https://github.com/libp2p/py-libp2p/blob/ddbedc6c154cff5c513ff149c0948c2394b0e6d8/libp2p/pubsub/pubsub.py#L398-L400